src/index.js

```diff
@@ -1,70 +1,699 @@
+        timeout: Infinity
+      }
+    )
+    return true
+  }
+
+  async getUserDataFromWebsite() {
+    this.log('info', '🤖 getUserDataFromWebsite')
+    if (
+      !(await this.isElementInWorker(
+        'a[href="/assurance/particuliers/vos-espaces-macif/espace-assurance/infos-persos"]'
+      ))
+    ) {
+      // With the created testAccount for the development, there is nothing to fetch at all
+      // No identity, no files whatsoever. Meaning that if the user just created his account and tries
+      // to launch the konnector, It would have failed on the next click.
+      // As there is no way of finish the execution successfully in this case, a clear error is needed.
+      throw new Error(
+        '❌️ This account has nothing to fetch, aborting execution'
+      )
+    }
+    await this.clickAndWait(
+      'a[href="/assurance/particuliers/vos-espaces-macif/espace-assurance/infos-persos"]',
+      'a[href="/assurance/particuliers/vos-espaces-macif/espace-assurance/infos-persos/modifier-email"]'
+    )
+    await this.runInWorkerUntilTrue({
+      method: 'checkInterceptions',
+      args: ['personnalInfos']
+    })
+    await this.runInWorker('getIdentity')
+    if (this.store.userIdentity) {
+      return { sourceAccountIdentifier: this.store.userIdentity.email }
+    } else {
+      throw new Error(
+        'No source account identifier found, the konnector should be fixed'
+      )
+    }
+  }
+
+  async fetch(context) {
+    this.log('info', '🤖 fetch')
+    // For now with the account we use to develop, it has been decide to not use FORCE_FETCH_ALL as the execution is pretty quick in every case
+    // Keeping this around for later use, when we'll get more use cases with other accounts.
+    // await this.determineForceFetchAll(context)
+    if (this.store.userCredentials) {
+      this.log('info', 'Saving credentials ...')
+      await this.saveCredentials(this.store.userCredentials)
+    }
+    if (this.store.userIdentity) {
+      this.log('info', 'Saving identity ...')
+      await this.saveIdentity({ contact: this.store.userIdentity })
+    }
+    await this.navigateToBillsPage()
+    await this.runInWorkerUntilTrue({
+      method: 'checkInterceptions',
+      args: ['attestations']
+    })
+    const { carAttestations, otherAttestations } = await this.runInWorker(
+      'getAttestations'
+    )
+    // Here we cannot use Promise.all because we're creating the subPath with both functions
+    // at the same time approx. leading to conflicts. Doing so, the subPath creation is done just once
+    await this.saveFiles(carAttestations, {
+      context,
+      contentType: 'application/pdf',
+      fileIdAttributes: ['filename'],
+      qualificationLabel: 'car_insurance',
+      subPath: 'Attestations'
+    })
+    await this.saveFiles(otherAttestations, {
+      context,
+      contentType: 'application/pdf',
+      fileIdAttributes: ['filename'],
+      // Cannot find an appropriate qualifications, has to be discuss
```

Comment:
Is the discussion ongoing ?

Reply:
Absolutely, i forgot to mention this in the daily.

I found no appropriate qualifications for the other attestation's types (other than car's, like house insurance, or person's insurance for accidents for example ...)

Reply:
Has been discuss, waiting for the appropriate version of the AA to add specific qualifications